src/config.rs

```diff
@@ -163,6 +190,65 @@ impl Config {
             None
         }
     }
+    pub(crate) fn discourse(&self) -> Option<Discourse> {
+        if let (Some(key), Some(user)) = (&self.discourse_api_key, &self.discourse_api_user) {
+            Some(Discourse::new(
+                "https://internals.rust-lang.org".to_owned(),
+                user.clone(),
+                key.clone(),
+            ))
+        } else {
+            None
+        }
+    }
+
+    pub(crate) fn blog_contents(
+        &self,
+        release: &str,
+        archive_date: &str,
+        for_blog: bool,
+        internals_url: Option<&str>,
+    ) -> Option<String> {
+        let human_date = self.scheduled_release_date.as_ref()?;
+        let release_notes_url = self.release_notes_url.as_ref()?;
+        let internals = internals_url
+            .map(|url| format!("You can leave feedback on the [internals thread]({url})."))
+            .unwrap_or_default();
+        let prefix = if for_blog {
+            format!(
+                r#"---
+layout: post
+title: "{} pre-release testing"
+author: promote-release
+team: The Release Team <https://www.rust-lang.org/governance/teams/release>
+---{}"#,
+                release, "\n\n",
+            )
+        } else {
+            String::new()
+        };
+        Some(format!(
+            "{prefix}The {release} pre-release is ready for testing. The release is scheduled for
+{human_date}. [Release notes can be found here.][relnotes]
+
+You can try it out locally by running:
+
+```plain
+RUSTUP_DIST_SERVER=https://dev-static.rust-lang.org rustup update stable
+```
+
+The index is <https://dev-static.rust-lang.org/dist/{archive_date}/index.html>.
+
+{internals}
+
+The release team is also thinking about changes to our pre-release process:
+we'd love your feedback [on this GitHub issue][feedback].
```

Comment:
I'd expect this might change between releases: it'd be nice to have the file in https://github.com/rust-lang/release-team/blob/master/pre-release-footer.md or something similar, so that the release team can effortlessly edit the text without having to mess with `promote-release`. Then we could fetch it and include it from `raw.githubusercontent.com`.

Reply:
I feel like adjusting the text here is pretty simple, I don't think having just a footer elsewhere is worthwhile. We're not really changing it often and when/if we do we're not unlikely to want to edit the body text too, so I don't think it's a win to split out the footer. The text is already generated in one place that's not heavily interspersed with noise, so is pretty easy to adjust IMO.

Reply:
Fair.